### DISCUSSION TITLE
Usage of formatstr from the DYNFMT and CALCCONV "apply" subcomands

### DISCUSSION THREAD

**Discussion by langbein-daniel** [dathere/qsv#1126] [Q&A] [answered]
Hi, I've downloaded the `qsv` binary from the GitHub releases. I tried the `apply` command both with the `dynfmt` and `calcconv` subcommands, but can't get a formatstring with a column-placeholder working.

My very basic CSV file:

```csv
FOO
1
```

The commands tried with their output:

```shell
qsv apply dynfmt --formatstr '{FOO} €' -c 'Total Price' "${file}"
```
```csv
FOO,Total Price
1,1
```

and

```shell
qsv apply calcconv --formatstr '{col1} + 10' --new-column BAR "${file}"
```
```csv
FOO,BAR
1,1
```

The expected/desired outputs for the second column would be `1 €` and `11`.

**Comment by ondohotola** [comment]
did you look at

qsv apply operations numtocurrency

**Comment by jqnatividad** [comment]
Hi @langbein-daniel ,
For the `dynfmt` example, this is because `dynfmt` sanitizes column names with a helper utility to ensure that all columns don't have special characters and are readily insertable into a SQL database.

This sanitization process also happens to fold column names to lowercase.

If you convert your example to use the "safe", sanitized column name in the `--formatstr` template:
```
qsv apply dynfmt --formatstr '{foo} €' -c 'Total Price' "${file}"
```
you will get the expected result:
```
FOO,Total Price
1, 1 €
```

As for the `calcconv` example, that's because you should have set formatstr to `{FOO} + 10`.

**Comment by jqnatividad** [comment, marked as answer]
For consistency and more intuitive behavior, I just made apply `DYNFMT` case sensitive.

https://github.com/jqnatividad/qsv/pull/1130

As for the calcconv example, that's because you should have set formatstr to `{FOO} + 10`, not `{col1} + 10`.
